docs/dev/api/index.rst

```diff
@@ -1,18 +1,351 @@
+   GridWorkflow.list_cells
+   GridWorkflow.list_tiles
+   GridWorkflow.load
+
+Grid Processing API Internals
+-----------------------------
+
+.. currentmodule:: datacube.api
+
+.. autosummary::
+   :toctree: generate/
+
+   GridWorkflow.cell_observations
+   GridWorkflow.group_into_cells
+   GridWorkflow.tile_sources
+
+Internal Data Model
+===================
+
+.. currentmodule:: datacube.model
+
+.. autosummary::
+   :toctree: generate/
+
+   Dataset
+   Measurement
+   MetadataType
+   DatasetType
+   GridSpec
+
+   Range
+   Variable
+   CellIndex
+
+
+
+Database Index API
+==================
+
+Dataset Querying
+----------------
+
+When connected to an ODC Database, these methods are available for
+searching and querying::
+
+   dc = Datacube()
+   dc.index.datasets.{method}
+
+.. currentmodule:: datacube.index._datasets.DatasetResource
+
+.. autosummary::
+   :nosignatures:
+
+   get
+   search
+   search_by_metadata
+   search_by_product
+   search_eager
+   search_product_duplicates
+   search_returning
+   search_summaries
+   has
+   bulk_has
+   can_update
+   count
+   count_by_product
+   count_by_product_through_time
+   count_product_through_time
+   get_derived
+   get_field_names
+   get_locations
+   get_archived_locations
+   get_datasets_for_location
+
+
+Dataset Writing
+---------------
+
+When connected to an ODC Database, these methods are available for
+adding, updating and deleting datasets::
+
+   dc = Datacube()
+   dc.index.datasets.{method}
+
+.. autosummary::
+   :nosignatures:
+
+   add
+   add_location
+   archive
+   archive_location
+   remove_location
+   restore
+   restore_location
+   update
+
+Product Querying
+----------------
+
+When connected to an ODC Database, these methods are available for
+discovering information about Products::
+
+   dc = Datacube()
+   dc.index.products.{method}
+
+.. currentmodule:: datacube.index._products.ProductResource
+
+.. autosummary::
+   :nosignatures:
+
+   from_doc
+   add
+   can_update
+   add_document
+   get
+   get_by_name
+   get_unsafe
+   get_by_name_unsafe
+   get_with_fields
+   search
+   search_robust
+   get_all
+
+Product Addition/Modification
+-----------------------------
+
+When connected to an ODC Database, these methods are available for
+discovering information about Products::
+
+   dc = Datacube()
+   dc.index.products.{method}
+
+.. currentmodule:: datacube.index._products.ProductResource
+
+.. autosummary::
+   :nosignatures:
+
+   from_doc
+   add
+   update
+   update_document
+   add_document
+
+Dataset to Product Matching
+---------------------------
+
+.. currentmodule:: datacube.index.hl
+
+.. autosummary::
+   :nosignatures:
+   :toctree: generate/
+
+   Doc2Dataset
+
+Geometry Utilities
+==================
+
+.. currentmodule:: datacube
+
+Open Data Cube includes a set of CRS aware geometry utilities.
+
+Geometry Classes
+----------------
+
+.. autosummary::
+   :nosignatures:
+   :toctree: generate/
+
+   utils.geometry.BoundingBox
```

Comment:
`BoundingBox` is not "CRS aware", and it's really annoying that it is the odd one out.